src/lib/RuntimeArgs.ts

```diff
@@ -34,20 +34,27 @@ export class NamedArg implements ToBytes {
     } = new CLStringBytesParser().fromBytesWithRemainder(bytes);
     const name = nameRes.unwrap();
     if (!nameRem) {
-      return resultHelper(Err('Missing data for value of named arg'));
+      return resultHelper<NamedArg, string>(
+        Err('Missing data for value of named arg')
+      );
     }
     const value = CLValueParsers.fromBytesWithType(nameRem).unwrap();
     return resultHelper(Ok(new NamedArg(name.value(), value)));
   }
 }
 
 const desRA = (_arr: any) => {
-  return new Map(
+  const parsed = new Map(
     Array.from(_arr, ([key, value]) => {
       const val = CLValueParsers.fromJSON(value);
       return [key, val.unwrap()];
     })
   );
+
+  if (parsed.size !== Array.from(_arr).length)
+    throw Error(`Duplicate key exists.`);
+
+  return parsed;
```

Comment:
in fact, this is acceptable to have two args with the same name - from node perspective they are represented as a vector of pairs, so for example if you have `[("a", 1), ("b", 2)]` `get_named_arg("a")` will return `1`. In our case we're stroring it as `Map` which prevents us from having duplicates. For now I think it's ok how it works, but we need to take info consideration that in v3 we will need to switch from native `Map` to our own implementation of list of pairs :).

Reply:
For about this, the [deploy on testnet](https://testnet.cspr.live/deploy/244d731044e1b7059507b9191a041844e93c8afa8789f03b5c5632ea85a7e286)
has 2 args for `store_tuple2` entry point.
But as you can see below
![image](https://user-images.githubusercontent.com/83718263/206896219-b9b932d0-aeb5-4c8b-a4ce-093f18bcc72e.png)
the values named `value` have two values: `valueforentrypointstore_tuple2` for string type and `123` for U512.
So we need to check if arguments shouldn't have duplicated keys.

Reply:
But this is no how the node acts. as I said node will accepts two arguments with the same name, it will just use the first one provided. in SDK we should replicate this behaviour, just wanted you to be aware about this, this will require use of list of pairs, instead of `Map`. But not for the 2.x.x, we will do this in 3.x